pilot/cmd/pilot-agent/main.go

```diff
@@ -152,12 +143,14 @@ var (
 			proxyConfig.ProxyAdminPort = int32(proxyAdminPort)
 			proxyConfig.Concurrency = int32(concurrency)
 
-			var pilotSAN []string
+			pilotSAN := make([]string, 0)
+			var ns string
+			setIdentityDomainAndDomain()
```

Comment:
The name of this func is very confusing. Please rename to something more meaningful.

Reply:
And it would be great to have some input/output values for testing purpose.